alns/accept/LateAcceptanceHillClimbing.py

```diff
@@ -0,0 +1,73 @@
+from collections import deque
+
+from alns.accept.AcceptanceCriterion import AcceptanceCriterion
+
+
+class LateAcceptanceHillClimbing(AcceptanceCriterion):
+    """
+    The Late Acceptance Hill Climbing (LAHC) criterion accepts a candidate
+    solution when it is better than the current solution from a number of
+    iterations before.
+
+    This implementation is based on the description of LAHC in [1].
+
+    Parameters
+    ----------
+    history_length: int
+        Non-negative integer specifying the maximum number of previous solutions
+        to be stored. Default: 0 (i.e., no objectives stored).
+    greedy: bool
+        Bool indicating whether or not to accept candidate solutions if they are
+        better than the current solution. Default: False.
+    collect_better: bool
+        Bool indicating whether or not to only collect current solutions that
+        are better than the previous current. Default: False
+
+    References
+    ----------
+    [1]: Burke, E. K., & Bykov, Y. The late acceptance hill-climbing heuristic.
+         *European Journal of Operational Research* (2017), 258(1), 70-78.
+    """
+
+    def __init__(
+        self,
+        history_length: int = 0,
+        greedy: bool = False,
+        collect_better: bool = False,
+    ):
+        self._history_length = history_length
+        self._greedy = greedy
+        self._collect_better = collect_better
+
+        if not isinstance(history_length, int) or history_length < 0:
+            raise ValueError("history_length must be a non-negative integer.")
+
+        self._objectives: deque = deque([], maxlen=history_length)
```

Comment:
Refactor: rename this to `_history`.

Reply:
Or `_obj_history`, if you prefer. But I do feel we need history in here somewhere.